### ISSUE TITLE
Condition matches_storage_class in the lifecycle_rules variable of the submodule simple_bucket must be an Array

### ISSUE THREAD

**Issue by odepedro-cr** [terraform-google-modules/terraform-google-cloud-storage#170]
### TL;DR

Condition matches_storage_class in the lifecycle_rules variable of the submodule simple_bucket must be an Array instead of string, like it is on the main.tf file of the main module.

[matches_storage_class = lookup(lifecycle_rule.value.condition, "matches_storage_class", null) ](https://github.com/terraform-google-modules/terraform-google-cloud-storage/blob/d217c7535a6745dcccd1dbf87524b11786627360/modules/simple_bucket/main.tf#L74)

[matches_storage_class      = contains(keys(lifecycle_rule.value.condition), "matches_storage_class") ? split(",", lifecycle_rule.value.condition["matches_storage_class"]) : null](https://github.com/terraform-google-modules/terraform-google-cloud-storage/blob/d217c7535a6745dcccd1dbf87524b11786627360/main.tf#L118)

### Expected behavior

Manage 'matches_storage_class' condition as a list of strings.

### Observed behavior

The simple_bucket submodule fails when using lifecycle rules because of this.

### Terraform Configuration

```hcl
N/A
```

### Terraform Version

```sh
Any
```

### Additional information

It was already reported here but there was no response.
https://github.com/terraform-google-modules/terraform-google-cloud-storage/issues/125

**Comment by pk0331** [comment]
It's an important for us, could you guys push the change.
